Sources/SourceKitLSP/Swift/FoldingRange.swift

```diff
@@ -127,61 +127,76 @@ fileprivate final class FoldingRangeFinder: SyntaxAnyVisitor {
   override func visitAny(_ node: Syntax) -> SyntaxVisitorContinueKind {
     if let braced = node.asProtocol(BracedSyntax.self) {
       return self.addFoldingRange(
-        start: braced.leftBrace.endPositionBeforeTrailingTrivia.utf8Offset,
-        end: braced.rightBrace.positionAfterSkippingLeadingTrivia.utf8Offset
+        start: braced.leftBrace.endPositionBeforeTrailingTrivia,
+        end: braced.rightBrace.positionAfterSkippingLeadingTrivia
       )
     }
     return .visitChildren
   }
 
   override func visit(_ node: ArrayExprSyntax) -> SyntaxVisitorContinueKind {
     return self.addFoldingRange(
-      start: node.leftSquare.endPositionBeforeTrailingTrivia.utf8Offset,
-      end: node.rightSquare.positionAfterSkippingLeadingTrivia.utf8Offset
+      start: node.leftSquare.endPositionBeforeTrailingTrivia,
+      end: node.rightSquare.positionAfterSkippingLeadingTrivia
     )
   }
 
   override func visit(_ node: DictionaryExprSyntax) -> SyntaxVisitorContinueKind {
     return self.addFoldingRange(
-      start: node.leftSquare.endPositionBeforeTrailingTrivia.utf8Offset,
-      end: node.rightSquare.positionAfterSkippingLeadingTrivia.utf8Offset
+      start: node.leftSquare.endPositionBeforeTrailingTrivia,
+      end: node.rightSquare.positionAfterSkippingLeadingTrivia
     )
   }
 
   override func visit(_ node: FunctionCallExprSyntax) -> SyntaxVisitorContinueKind {
-    return self.addFoldingRange(
-      start: node.arguments.position.utf8Offset,
-      end: node.arguments.endPosition.utf8Offset
-    )
+    let start = node.leftParen?.endPositionBeforeTrailingTrivia ?? node.arguments.position
+    let end =
+      if !node.additionalTrailingClosures.isEmpty {
+        node.additionalTrailingClosures.endPositionBeforeTrailingTrivia
+      } else if let trailingClosure = node.trailingClosure {
+        trailingClosure.endPositionBeforeTrailingTrivia
+      } else if let rightParen = node.rightParen {
+        rightParen.positionAfterSkippingLeadingTrivia
+      } else {
+        // Should never happen because the call should have either a trailing closure or a closing ')'
```

Comment:
Could just have a call that doesn't have it's closing `)` yet couldn't we?

Reply:
In that case the parser would have synthesized a missing `)`.